src/helics/common/JsonProcessingFunctions.cpp

```diff
@@ -16,15 +17,54 @@ SPDX-License-Identifier: BSD-3-Clause
 
 namespace helics::fileops {
 
-bool looksLikeCommandLine(std::string_view testString)
+static std::string_view removeSpaceAndComment(std::string_view jsonString)
 {
-    if (testString.empty()) {
+    gmlc::utilities::string_viewOps::trimString(jsonString);
+    while (jsonString.size() > 2) {
+        if (jsonString[0] == '/' && jsonString[1] == '/') {
+            auto nextNewLine = jsonString.find_first_of('\n');
+            if (nextNewLine == std::string_view::npos) {
+                return {};
+            }
+            jsonString.remove_prefix(nextNewLine + 1);
+            gmlc::utilities::string_viewOps::trimString(jsonString);
+        } else {
+            break;
+        }
+    }
+    gmlc::utilities::string_viewOps::trimString(jsonString);
+    return jsonString;
+}
+
+bool looksLikeConfigJson(std::string_view jsonString)
+{
+    if (jsonString.find("\n#") != std::string_view::npos) {
         return false;
     }
-    if (testString.front() == '-') {
-        return true;
+    jsonString = removeSpaceAndComment(jsonString);
+    if (jsonString.size() < 7) {
+        // minimum viable config file is 7 characters {"f":4}
+        return false;
     }
-    return testString.find(" -") != std::string::npos;
+
+    if (jsonString.front() != '{') {
+        return false;
+    }
+    auto firstQuote = jsonString.find_first_of("\"'");
+    if (firstQuote == std::string_view::npos) {
+        return false;
+    }
+    auto firstColonLoc = jsonString.find_first_of(':');
+    if (firstColonLoc == std::string_view::npos) {
+        return false;
+    }
+    auto closeBracket = jsonString.find_last_of('}');
+    if (closeBracket == std::string_view::npos) {
+        return false;
+    }
+
+    auto afterBracket = removeSpaceAndComment(jsonString.substr(closeBracket + 1));
```

Comment:
I think this could pick up the wrong closing bracket if there are trailing comment lines and one happens to be something like `// ... }`. But this is probably good enough as a heuristic.